### PR TITLE
[SPARK-28055][SS][DSTREAMS] Add delegation token custom AdminClient configurations.

### PR DESCRIPTION
**Pull request by gaborgsomogyi**
## What changes were proposed in this pull request?

At the moment Kafka delegation tokens are fetched through `AdminClient` but there is no possibility to add custom configuration parameters. In [options](https://spark.apache.org/docs/2.4.3/structured-streaming-kafka-integration.html#kafka-specific-configurations) there is already a possibility to add custom configurations.

In this PR I've added similar this possibility to `AdminClient`.

## How was this patch tested?

Existing + added unit tests.

```
cd docs/
SKIP_API=1 jekyll build
```
Manual webpage check.

### CODE REVIEW

**Comment by gaborgsomogyi** [comment]
@koertkuipers the feature is on the way what we've spoken about. Ping me if you think this area lacks something.

**Comment by koertkuipers** [comment]
looks good to me

**Comment by gaborgsomogyi** [comment]
cc @vanzin

File: docs/structured-streaming-kafka-integration.md [hunk 1]
```diff
@@ -819,6 +819,11 @@ Delegation tokens can be obtained from multiple clusters and <code>${cluster}</c
   </tr>
 </table>
 
+#### Kafka Specific Configurations
+
+Kafka's own configurations can be set with `kafka.` prefix, e.g, `--conf spark.kafka.clusters.${cluster}.kafka.retries=1`.
+For possible kafka parameters, see [Kafka adminclient config docs](http://kafka.apache.org/documentation.html#adminclientconfigs).
```
Comment (vanzin) [inline docs/structured-streaming-kafka-integration.md]:
```suggestion
For possible Kafka parameters, see [Kafka adminclient config docs](http://kafka.apache.org/documentation.html#adminclientconfigs).
```

File: external/kafka-0-10-token-provider/src/main/scala/org/apache/spark/kafka010/KafkaTokenSparkConf.scala [hunk 2]
```diff
@@ -47,7 +48,8 @@ private[spark] case class KafkaTokenClusterConf(
     s"keyStoreLocation=$keyStoreLocation, " +
     s"keyStorePassword=${keyStorePassword.map(_ => "xxx")}, " +
     s"keyPassword=${keyPassword.map(_ => "xxx")}, " +
-    s"tokenMechanism=$tokenMechanism}"
+    s"tokenMechanism=$tokenMechanism, " +
+    s"specifiedKafkaParams=$specifiedKafkaParams}"
```
Comment (vanzin) [inline external/kafka-0-10-token-provider/src/main/scala/org/apache/spark/kafka010/KafkaTokenSparkConf.scala]: Do these need to be redacted?

**Comment by koertkuipers** [comment]
this branch works for me to set `ssl.endpoint.identification.algorithm` to empty string for admin client

File: external/kafka-0-10-token-provider/src/main/scala/org/apache/spark/kafka010/KafkaTokenSparkConf.scala (hunk 2 above)
Comment (gaborgsomogyi) [reply external/kafka-0-10-token-provider/src/main/scala/org/apache/spark/kafka010/KafkaTokenSparkConf.scala]: Yep, that't a nice catch. Fixing...

**Comment by gaborgsomogyi** [comment]
@koertkuipers Thanks for testing it!

**Comment by vanzin** [comment]
retest this please

**Comment by vanzin** [comment]
Merging to master.

**Pull request closed without merging**